tracing/otel.go

```diff
@@ -185,6 +185,10 @@ func MaybeJaegerRemoteSamplerFromEnv(serviceName string) (tracesdk.Sampler, bool
 		return nil, false, nil
 	}
 
+	// Unset the OTEL_TRACES_SAMPLER environment variable to the SDK's samplerFromEnv()
+	// function complaining about unknown sampler and logging confusing messages.
+	_ = os.Unsetenv("OTEL_TRACES_SAMPLER")
```

Comment:
Nit: Since this unsets the env var that it reads just prior, is it worth documenting that `MaybeJaegerRemoteSamplerFromEnv` might have different effects if it's called multiple times?